conf.d/forgit.plugin.fish

```diff
@@ -27,206 +27,27 @@ set | awk -F ' ' '{ print $1 }' | grep FORGIT_ | while read var
     end
 end
 
-function forgit::log -d "git commit viewer"
-    "$FORGIT" log $argv
-end
-
-function forgit::diff -d "git diff viewer" --argument-names arg1 arg2
-    "$FORGIT" diff $argv
-end
-
-function forgit::add -d "git add selector" --wraps "git add"
-    "$FORGIT" add $argv
-end
-
-function forgit::reset::head -d "git reset HEAD (unstage) selector"
-    "$FORGIT" reset_head $argv
-end
-
-function forgit::stash::show -d "git stash viewer"
-    "$FORGIT" stash_show $argv
-end
-
-function forgit::stash::push -d "git stash push selector" ()
-    "$FORGIT" stash_push $argv
-end
-
-function forgit::clean -d "git clean selector"
-    "$FORGIT" clean $argv
-end
-
-function forgit::cherry::pick -d "git cherry-picking" --argument-names 'target' --wraps "git cherry-pick"
-    "$FORGIT" cherry_pick $argv
-end
-
-function forgit::cherry::pick::from::branch -d "git cherry-picking with interactive branch selection" --wraps "git cherry-pick"
-    "$FORGIT" cherry_pick_from_branch $argv
-end
-
-function forgit::rebase -d "git rebase"
-    "$FORGIT" rebase $argv
-end
-
-function forgit::fixup -d "git fixup"
-    "$FORGIT" fixup $argv
-end
-
-function forgit::checkout::file -d "git checkout-file selector" --argument-names 'file_name' --wraps "git checkout --"
-    "$FORGIT" checkout_file $argv
-end
-
-function forgit::checkout::branch -d "git checkout branch selector" --argument-names 'input_branch_name' --wraps "git branch"
-    "$FORGIT" checkout_branch $argv
-end
-
-function forgit::checkout::tag -d "git checkout tag selector" --argument-names 'tag_name' --wraps "git checkout"
-    "$FORGIT" checkout_tag $argv
-end
-
-function forgit::checkout::commit -d "git checkout commit selector" --argument-names 'commit_id' --wraps "git checkout"
-    "$FORGIT" checkout_commit $argv
-end
-
-function forgit::branch::delete -d "git branch deletion selector" --wraps "git branch --delete"
-    "$FORGIT" branch_delete $argv
-end
-
-function forgit::revert::commit -d "git revert commit selector" --argument-names 'commit_hash' --wraps "git revert --"
-    "$FORGIT" revert_commit $argv
-end
-
-function forgit::blame -d "git blame viewer"
-    "$FORGIT" blame $argv
-end
-
-function forgit::ignore -d "git ignore generator"
-    "$FORGIT" ignore $argv
-end
-
-function forgit::ignore::update
-    "$FORGIT" ignore_update $argv
-end
-
-function forgit::ignore::get
-    "$FORGIT" ignore_get $argv
-end
+# Add `git-forgit` command to PATH
+fish_add_path -ga "$FORGIT_INSTALL_DIR/bin"
```

Comment:
Is there any way to do this without affecting the user's $PATH? That seems like it could be a little invasive?

Reply:
I can replace `git-forgit` in the abbreviations with `$FORGIT`. Completion shouldn't be affected, but the expanded command line might look a bit ugly (fish expands the full path of `$FORGIT_INSTALL_DIR/bin/git-forgit`.

Reply:
Meaning when you expand the abbreviation? 

It would expand:
```
ga
```
becomes
```
$FORGIT_INSTALL_DIR/bin/git-forgit add
```

For the user? 

Instead of 

```
git forgit add
```

Reply:
I added `git-forgit` alias to the full-path command, so hopefully it'll solve the "lengthy command line" problem.

Reply:
Great! Let me try this out for a bit and get a feel for it before merging. Overall code structure wise it's a big improvement, just going to perform some testing to make sure I understand it and there aren't any gotchas.

Reply:
Still testing this! It's definitely a change as a user, but easy to adapt to. It also de-duplicates the code so much, which is a huge positive.